cherryontech/unmatched
# #UM-33: Home page responsiveness

## Description
This PR fixes the responsiveness of the home page for tablet and mobile views. Currently, this PR has adjusted responsiveness of the homepage carousel slide 1 ONLY and has adjusted the image on that carousel (which was previously stretched/distorted). This PR also has refactored the components using React's philosophy of separation of concerns to make sure each component is rendered in the most logical place and that no component is too big/unwieldy and is made of smaller components instead (specifically the Hero Component). 


## Related Tickets & Documents
This ticket closes #UM-33


## Acceptance Criteria



- [ ] Ensure the homepage adheres to Tailwind CSS breakpoints sm, md, lg breakpoints
- [ ] The homepage layout must adjust content, including the carousel, text blocks, and profile grids, to suit the screen size without compromising functionality or aesthetics.
- [ ] The hero carousel should display correctly on all devices, with navigation arrows and pagination dots remaining interactive and accessible.
- [ ] All text, including headings, paragraphs, and button labels, should maintain readability across devices, with size and spacing adjustments as needed.
- [ ] Navigation elements, such as the menu bar and buttons, should remain functional and easy to interact with, even on smaller screens.
- [ ] On a laptop, display all profiles in a row. On a tablet and on a phone, stack profiles in a single column format.
- [ ] Ensure that all interactive elements, including buttons and links, are easily tappable on touchscreen devices with appropriate size and spacing.

## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|     | :sparkles: New feature     |
| ✓ | 🎨 Style                   |
|  ✓ | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |


## Mobile & Desktop Screenshots/Recordings

**BEFORE** Desktop:
<img src="https://github.com/cherryontech/unmatched/assets/115492619/a450d050-fcd1-4734-a9c2-ec8578a6c4d5" width="400">
**AFTER** Desktop:
<img src="https://github.com/cherryontech/unmatched/assets/115492619/96d0eef9-0c86-4fa7-9b69-d090901538b3" width="400">

**BEFORE** Tablet:
<img src="https://github.com/cherryontech/unmatched/assets/115492619/a4dc8122-abc7-4e4c-bc77-3e0651b31beb" width="300">
**AFTER** Tablet:
<img src="https://github.com/cherryontech/unmatched/assets/115492619/c8190068-7dae-43ae-bd9b-99a9132d22df" width="300">

**BEFORE** Mobile:
<img src="https://github.com/cherryontech/unmatched/assets/115492619/ebd71688-dc62-4dd1-9b2e-1e4d628460c7" width="250">
**AFTER** Mobile:
<img src="https://github.com/cherryontech/unmatched/assets/115492619/5751507b-4e9c-493e-8fc6-932fa632609f" width="250">



## What I learned
Looking at the whole home page as one component and trying to address responsiveness is overwhelming and ineffective. Instead I broke carousel slide 1 off into a separate component and started looking at the responsive of each element in that component, starting with the hero container div, moving to the image size, and then looking at text size and positioning. Also Tailwind properties like `object-fit` and `object-cover` are needed to maintain an image's proportions/ratios

## Testing steps
1. Switch to this branch with the following command: `git switch style/UM-33-home-responsive`
2. Run the following scripts: `npm run dev` to start Vite's dev build & `npm run tailwind` to start the Tailwind CSS build
3. View the app in browser (the `npm run dev` script will return the url you can see the app at)
4. Inspect the home page in relation to the Figma design (using Chrome Dev Tools or similar to view different screen sizes). NOTE: use the iPad Air option for tablet vs iPad Pro as the iPad Pro defaults to the 'large' breakpoint


## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
| ✓  | 🙅 no documentation needed    |


## What gif best describes this PR or how it makes you feel?
  ![shimmy](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExeTZqa2szcnh0bXV1dmdvdnBtNDM1a25kMWZvZnloM3A4dGw4NHI4ayZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/nNxT5qXR02FOM/giphy.gif)
